### PR TITLE
Add Docker integration tests (#141)

### PR DESCRIPTION
**Pull request by tboeghk**
This PR add basic Docker integration tests. The Querqy `jar-with-dependencies` is built and tested against all minor Versions in the Solr 8.x release branch. Solr is populated with the [chorus ecommerce](https://github.com/querqy/chorus/tree/master/solr/configsets/ecommerce/conf) and fed with [~19k products from a icecat excerpt](https://space.thiswayup.de/icecat-products-w_price-19k-20201127.tgz). The querqy functionality is tested agains the query `laptop` for which a common rule set is defined.

The integration tests are disabled by default. To enable them, pass the `-DskipITs=false` flag to Maven:

```
mvn -DskipITs=false verify
```

### CODE REVIEW

**Comment by tboeghk** [comment]
I'll furthermore add the integration-tests to Github Actions.

To meet the JDK tests, I'll add compile & test for all recent JDKs 8-15.

**Comment by renekrie** [comment]
Fantastic! Thank you @tboeghk 

I'll change the file name and point to the querqy.org server to avoid the external dependency.

**Comment by tboeghk** [comment]
I updated the Github Actions flow, so this PR is good to merge 🥇

**Comment by tboeghk** [comment]
Hmmmm, some tests seem to be flaky :-/

**Comment by tboeghk** [comment]
Fixed it with a 3 second wait until the cache is loaded in Solr. This caused the flaky test: 640532c9dc6f6ad587b3e2595fddea32520c6ac2

**Comment by tboeghk** [comment]
I updated the Github Actions flow, so this PR is good to merge 🥇 **lol**

**Comment by epugh** [comment]
This is really impressive @tboeghk piece of code.    I'm hoping I can save this PR for the future as a model for this.    Are you going to write this up as a blog?    If not, let me know, and if you're cool with it, I'd write a "What I learned from Torsten's PR #147 " style blog.    This is the kind of pattern that we need to support many of the Solr packages over time!    I wonder if you could extract anything from this to make it more repeatable for the next Package?    Or is all the plumbing you did kind of specific to Querqy.

File: querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/QuerqySolrClientUtils.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import org.eclipse.jetty.client.HttpClient;
+import org.testcontainers.containers.SolrClientUtils;
+
+/**
+ * Mostly copied code from {@link SolrClientUtils} as it's not open for
+ * extension :-/
+ */
+public class QuerqySolrClientUtils extends SolrClientUtils {
+
+    private static final String CHORUS_DATASET_URL = "https://querqy.org/datasets/icecat/icecat-products-w_price-19k-20201127.tar.gz";
```
Comment (renekrie) [inline querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/QuerqySolrClientUtils.java:29]: It seems that this var is not used.

File: querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/QuerqySolrClientUtils.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.solr.client.solrj.SolrClient;
```
Comment (renekrie) [inline querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/QuerqySolrClientUtils.java:18]: Are we still using this? - looks like we just use HttpClient

File: querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/SolrQuerqyIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        SolrPingResponse response = new SolrPing().process(client, QuerqySolrContainer.QUERQY_IT_COLLECTION_NAME);
+        MatcherAssert.assertThat(response.getStatus(), Matchers.is(0));
+
+        // querqy rules get applied
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.set(CommonParams.Q, "laptop");
+        params.set("defType", "querqy");
+        params.set(DisMaxParams.QF, "name title product_type short_description ean search_attributes");
+        QueryResponse query = client.query(QuerqySolrContainer.QUERQY_IT_COLLECTION_NAME, params);
+        MatcherAssert.assertThat(query.getResults().getNumFound(), Matchers.is(42L));
```
Comment (renekrie) [inline querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/SolrQuerqyIntegrationTest.java:69]: 42L? How did you do that? ;-)

**Review by renekrie — CHANGES_REQUESTED** [review]
I've marked up minor issues where it seems that some unused code could be removed

File: querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/SolrQuerqyIntegrationTest.java (hunk 3 above)
Comment (tboeghk) [reply querqy-for-lucene/querqy-solr/src/test/java/querqy/solr/it/SolrQuerqyIntegrationTest.java:69]: pure math magic 🔢

**Pull request merged**